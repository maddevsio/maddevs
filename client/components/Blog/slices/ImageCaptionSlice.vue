<template>
  <div>
    <div class="post-part single">
      <div :class="size" class="block-img">
        <prismic-image :field="img" :class="{ 'block-img-zoom': zoomEnabled }" @click="openModal" />
        <SimpleModal v-if="zoomEnabled" ref="zoom">
          <prismic-image :field="img" />
        </SimpleModal>
      </div>
      <p v-if="caption" class="image-label">{{ caption }}</p>
    </div>
  </div>
</template>

<script>
import SimpleModal from '@/containers/SimpleModal'

export default {
  name: 'ImageCaptionSlice',
  components: { SimpleModal },
  props: {
    slice: {
      type: Object,
      required: true,
    },
  },

  data() {
    return {
      img: '',
      caption: '',
      size: '',
    }
  },

  computed: {
    zoomEnabled() {
      return this.slice.primary.enable_zoom === 'enable'
    },

    lightBoxImage() {
      return {
        src: this.img.url,
        alt: this.img.alt,
      }
    },
  },

  created() {
    this.img = this.slice.primary.image
    this.caption = this.$prismic.asText(this.slice.primary.caption)
    this.size = this.slice.slice_label
  },

  methods: {
    openModal() {
      if (this.slice && this.slice.primary.enable_zoom === 'enable') {
        this.$refs.zoom.show()
      }
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../../../assets/styles/_vars';

img {
  max-width: 100%;
  height: auto;
  vertical-align: middle;
}

.block-img {
  margin-bottom: 10px;
  text-align: center;

  &-zoom {
    cursor: pointer;
  }
}

.image-label {
  display: block;
  text-align: center;
  font-size: 14px;
  line-height: 129%;
  letter-spacing: -0.02em;
  font-family: 'Poppins-Regular', sans-serif;
  color: $text-color--grey;
  margin-bottom: 12px;
}
</style>
