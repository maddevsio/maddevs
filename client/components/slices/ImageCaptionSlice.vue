<template>
  <div>
    <div class="post-part single">
      <div
        :class="size"
        class="block-img"
      >
        <img
          class="img_lazy"
          :class="{ 'block-img-zoom': zoomEnabled }"
          :data-src="img.url"
          :alt="img.alt"
          width="632"
          height="170"
          @click="openModal"
        >
        <SimpleModal
          v-if="zoomEnabled"
          ref="zoom"
        >
          <img
            :src="img.url"
            :alt="img.alt"
          >
        </SimpleModal>
      </div>
      <p
        v-if="caption"
        data-testid="test-image-label"
        class="image-label"
      >
        {{ caption }}
      </p>
    </div>
  </div>
</template>

<script>
import SimpleModal from '@/components/slices/SimpleModal'

export default {
  name: 'ImageCaptionSlice',
  components: { SimpleModal },
  props: {
    slice: {
      type: Object,
      required: true,
    },
  },

  data() {
    return {
      img: '',
      caption: '',
      size: '',
    }
  },

  computed: {
    zoomEnabled() {
      return this.slice.primary.enable_zoom === 'enable'
    },
  },

  created() {
    this.img = this.slice.primary.image
    this.caption = this.$prismic.asText(this.slice.primary.caption)
    this.size = this.slice.slice_label
  },

  methods: {
    openModal() {
      if (this.slice && this.slice.primary.enable_zoom === 'enable') {
        this.$refs.zoom.show()
      }
    },
  },
}
</script>

<style lang="scss" scoped>
.block-img {
  margin-bottom: 10px;
  text-align: center;
  &-zoom {
    cursor: pointer;
  }
}

.image-label {
  @include font('Inter', 14px, 400);
  display: block;
  text-align: center;
  line-height: 129%;
  letter-spacing: -0.02em;
  color: $text-color--grey;
  margin-bottom: 12px;
}
</style>
