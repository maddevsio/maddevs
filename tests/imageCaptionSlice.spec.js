import { mount } from '@vue/test-utils'
import ImageCaptionSlice from '../client/components/Blog/slices/ImageCaptionSlice'

describe('image attribute slice component', () => {
  let wrapper

  const slice = {
    items: [{}],
    primary: {
      image: {
        alt: 'alt text',
        copyright: null,
        dimensions: { height: 298, width: 1200 },
        url:
          'https://images.prismic.io/superpupertest/4b27a325-c48e-4f20-943d-e40537291055_2020-12-08_14-11.png?auto=compress,format&rect=0,0,1231,306&w=1200&h=298',
      },
      caption: 'caption',
    },
    slice_label: 'image-full-width',
    slice_type: 'image_with_caption',
  }

  beforeEach(() => {
    wrapper = mount(ImageCaptionSlice, {
      propsData: { slice },
      mocks: {
        $prismic: {
          asText: text => text,
          asHtml: html => html,
        },
      },
      stubs: ['prismic-image'],
    })
  })

  it('is vue component', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('caption rendered', () => {
    expect(wrapper.find('.image-label').text()).toBe(slice.primary.caption)
  })
})
