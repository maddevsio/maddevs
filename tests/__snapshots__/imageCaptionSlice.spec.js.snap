// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image attribute slice component renders correctly 1`] = `
<div>
  <div
    class="post-part single"
  >
    <div
      class="block-img image-full-width"
    >
      <prismic-image-stub
        class=""
        field="[object Object]"
      />
       
      <!---->
    </div>
     
    <p
      class="image-label"
    >
      caption
    </p>
  </div>
</div>
`;
